tracer/build/_build/Build.Steps.cs

```diff
@@ -2695,6 +2733,17 @@ static LogLevel ParseNativeLogLevel(string value)
             (false, true) => ($"linux-musl-{UnixArchitectureIdentifier}", "so"),
         };
 
+    private (string Arch, string Ext) GetWinArchitectureAndExtension(string fmk)
+    {
+        return RuntimeInformation.ProcessArchitecture switch
+        {
+            Architecture.X86 => ("win-x86", "dll"),
+            Architecture.X64 => ("win-x64", "dll"),
+            Architecture.Arm64 => ("win-arm64", "dll"),
+            _ => ("win-x86", "dll") // Default to x86 for unknown architectures
```

Comment:
Does `RuntimeInformation.ProcessArchitecture` reflect the correct target? Don't we end up using the same build process for multiple archs?

Reply:
🤔 Hmmm, this doesn't look right to me 🤔 

Specifically "gives the arch of the currently running test host app" is not correct. This _always_ gives x64, because we always run _Nuke_ as an x64 app on Windows (as we don't currently build for arm64). We _subsequently_ run with `DOTNET_EXE_32` when we're doing the _actual_ `dotnet test` part (controlled by `SetDotnetPath`);

Also, nit: we don't need to return `dll` if it's always dll 🤷‍♂️